### PR TITLE
Some fix to macros docs (#7608)

### PR DESCRIPTION
**Pull request by Blaisorblade**
Some changed needed for #7608. I'm sure more such changes are needed; I had to do this to answer some questions on Gitter, but I don't have time to go through the entire document.

### CODE REVIEW

File: docs/docs/reference/metaprogramming/macros.md [hunk 1]
```diff
@@ -36,12 +36,12 @@ import scala.quoted._
 inline def assert(expr: => Boolean): Unit =
   ${ assertImpl('expr) }
 
-def assertImpl(expr: Expr[Boolean]) = '{
+def assertImpl(expr: Expr[Boolean])(given qctx: QuoteContext) = '{
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/macros.md]:
```suggestion
def assertImpl(expr: Expr[Boolean])(given QuoteContext) = '{
```

File: docs/docs/reference/metaprogramming/macros.md [hunk 2]
```diff
@@ -36,12 +36,12 @@ import scala.quoted._
...
   if (!$expr)
     throw new AssertionError(s"failed assertion: ${${ showExpr(expr) }}")
 }
 
-def showExpr(expr: Expr[Boolean]): Expr[String] =
+def showExpr(expr: Expr[Boolean])(given qctx: QuoteContext): Expr[String] =
```
Comment (nicolasstucki) [inline docs/docs/reference/metaprogramming/macros.md]:
```suggestion
def showExpr(expr: Expr[Boolean])(given QuoteContext): Expr[String] =
```

**Comment by nicolasstucki** [comment]
Thanks @Blaisorblade

**Comment by Blaisorblade** [comment]
Thanks for the quick merge! 😄

**Pull request merged**